src/ui/display.c

```diff
@@ -259,6 +259,13 @@ bool skip_change_outputs() {
     inplace_selection_sort((size_t) G_context.tx_info.change_len, change_indices);
 
     for (uint8_t i = 0; i < G_context.tx_info.change_len; i++) {
+        // current_output is the number of parsed outputs
+        // If it is the same as confirmed_outputs we have confirmed all parsed outputs
+        if (G_context.tx_info.confirmed_outputs >= G_context.tx_info.current_output) {
```

Comment:
question(non-blocking): Why did you add this verification inside the `for` loop? It doesn't use the index, so maybe we could move it before. What do you think?

Reply:
Moved the check to outside the loop.